### ISSUE TITLE
Wrong $ref for path in nested dir processing in v1

### ISSUE THREAD

**Issue by ZsergUA** [swagger-api/swagger-parser#1223]
In case `$ref` for `path` is pointing to a file in nested directory, internally defined `ref`s are using wrong base directory.

While having `swagger.yaml` with:
```
paths:
  /some_path:
    $ref: './subdir/some.yaml'
```
and `some.yaml` with:
```
get:
  responses:
    200:
      schema:
        type: object
        properties:
          some_property:
            $ref: './some_definitions.yaml#/SomeProperty'
```
will try to resolve `some_definitions.yaml` in the `swagger.yaml` root directory instead of `subdir`

**Comment by ZsergUA** [comment]
pull request with fix #1224

**Comment by gracekarina** [comment]
thanks! @ZsergUA 
the PR is now merged.
